### ISSUE TITLE
Keras/Tensorflow

### ISSUE THREAD

**Issue by trevorlapay** [trevorlapay/UNMFall18_ML_Project3#1]
I've been playing around with keras and tensorflow using the data generated from wav files and the fft method specified by Trilce in the project doc. I still need to go through the keras documentation to fully understand how to build a good model, but at the very least, this is a framework for generating models and running them in python. Right now the accuracy is hideously low so this is purely didactic for now, but it's progress.

**Comment by trevorlapay** [comment]
I forgot to mention the issue: Tensorflow is not compatible with python 3.7, at least in my setup. I'm running successfully now using the Anaconda with python 3.6 so I can leverage my GPU, but it was a massive headache trying to figure out why tensorflow wasn't working with 3.7. So, if you are using 3.7, keep this in mind

**Comment by trevorlapay** [comment]
Getting around 38% accuracy now that I understand tensorflow a bit better with a out of the box neural network It sounds like we'll have to use a convolutional (1D? 2D) NN to get high results based on what I read around the internet.  I'll need to read a lot more about CNNs to fully understand what's going on with them -- they appear ideal for image recognition, but lots of people are using them for music. I wonder if they're pushing the spectrograph images themselves through the CNN...

**Comment by trevorlapay** [comment]
Now at 65%. I think with some tweaking I can make this work even without using cnns. I added a bunch of code for MFCC, but I'm having trouble getting it to predict. For some reason, model.predict() is returning the training set instead of the one hot category arrays. (I swear it was working earlier, but I don't know what happened. Might be by environment?) Someone please take a look. To switch to MFCC, toggle the ceps=True argument in the wav loader for the training and test files.

**Comment by trevorlapay** [comment]
Good news - CEPS now works (35% acc, which is expected).The accuracy we get with frequency/amplitude after tuning it for a while is 67%, which is within Trilce's threshold, so we're done with that part. We have 2 remaining things to do:

1) One more feature extraction method (Spectrogram CNN?)
2) One more classifier.

Both of these can get knocked out by doing the spectrogram if you use Conv2D in keras, and you don't need to stress out about accuracy.

That, and the report. Looking good!

**Comment by hankyusa** [comment]
Oh I was just looking at CEPS. Oh well I should be familiar with what you did anyway. You've done nearly all that work, Trevor, but I promise this weekend is nothing but machine learning for me.

**Comment by trevorlapay** [comment]
Sounds good. I'll start the report for the stuff pertaining to what I did

**Comment by trevorlapay** [comment]
I took a stab at the confusion matrix and loaded those plots into the report.
